core/src/main/scala/akka/kafka/javadsl/ElementProducer.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <https://softwaremill.com>
+ * Copyright (C) 2016 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.kafka.javadsl
+
+import java.util.concurrent.{CompletionStage, Executor}
+
+import akka.kafka.ProducerMessage._
+import akka.kafka.{scaladsl, ProducerSettings}
+import org.apache.kafka.clients.producer.{ProducerRecord, RecordMetadata}
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext
+
+/**
+ * Utility class for producing to Kafka without using Akka Streams.
+ */
+final class ElementProducer[K, V] private (underlying: scaladsl.ElementProducer[K, V]) extends AutoCloseable {
+
+  /**
+   * Utility class for producing to Kafka without using Akka Streams.
+   * @param settings producer settings used to create or access the [[org.apache.kafka.clients.producer.Producer]]
+   */
+  def this(settings: ProducerSettings[K, V], ec: Executor) =
+    this(scaladsl.ElementProducer(settings)(ExecutionContext.fromExecutor(ec)))
+
+  /**
+   * Utility class for producing to Kafka without using Akka Streams.
+   * @param settings producer settings used to create or access the [[org.apache.kafka.clients.producer.Producer]]
+   */
+  def this(settings: ProducerSettings[K, V], ec: ExecutionContext) =
```

Comment:
a dispatcher is:
```
trait ExecutionContextExecutor extends ExecutionContext with Executor
```

which of the constructors will be choosen for that? e.g. `system.getDispatcher()`

Reply:
Thanks, that did not work as javac couldn't decide. I'll remove the `ExecutionContext` variant.